services/community-site/src/pages/Delegation/NodePage.tsx

```diff
@@ -0,0 +1,313 @@
+
+import './nodePage.scss';
+import Delegate from './Modal/Delegate';
+import Undelegate from './Modal/Undelegate';
+import CloseIcon from '../../assets/icons/close';
+
+const NodePage = () => {
+  const auth = useAuth();
+  const { status, account } = useMetaMask();
+  const [availableBalance, setAvailableBalance] = useState(0);
+  const [delegationAtTop, setDelegationAtTop] = useState<boolean>(false);
+  const [node, setNode] = useState<PublicNode | null>(null);
+  const [delegationCount, setDelegationCount] = useState<number>(0);
+  const [delegations, setDelegations] = useState<Delegation[] | []>([]);
+  const [delegationPage, setDelegationPage] = useState<number>(1);
+  const [delegateToNode, setDelegateToNode] = useState<PublicNode | null>(null);
+  const [undelegateFromNode, setUndelegateFromNode] = useState<PublicNode | null>(null);
+  const { nodeId } = useParams<{ nodeId?: string }>();
+  const delegationApi = useDelegationApi();
+  const nodeIcon = toSvg('Aweesome node 1', 40, { backColor: '#fff' });
+
+  const isLoggedIn = !!auth.user?.id;
+  const canDelegate = isLoggedIn && status === 'connected' && !!account;
+
+  const getBalances = useCallback(async () => {
+    if (!canDelegate) {
+      return;
+    }
+    const data = await delegationApi.get(`/delegations/${account}/balances`);
+    if (data.success) {
+      setAvailableBalance(data.response.remaining);
+    }
+  }, [canDelegate, account]);
+
+  const fetchNode = useCallback(async () => {
+    const data = await delegationApi.get(`/validators/${nodeId}`);
+    if (data.success) {
+      setNode(data.response);
+      // eslint-disable-next-line no-console
+      console.log(data.response);
+      // eslint-disable-next-line no-console
+      console.log(node);
+    }
+  }, [nodeId]);
+
+  const fetchDelegators = useCallback(async () => {
+    setDelegations([]);
+    const data = await delegationApi.get(
+      `/validators/${nodeId}/delegations?show_my_delegation_at_the_top=${delegationAtTop}&page=${delegationPage}`,
+    );
+    if (data.success) {
+      setDelegations(data.response.data);
+      setDelegationCount(data.response.count);
+    }
+  }, [nodeId, delegationAtTop, delegationPage]);
+
+  useEffect(() => {
+    getBalances();
+    fetchNode();
+    fetchDelegators();
+  }, [fetchNode, fetchDelegators]);
+
+  const delegationPossible = (node?.totalDelegation || 0) + (node?.remainingDelegation || 0);
+  const communityDelegated =
+    (((node?.totalDelegation || 0) - (node?.ownDelegation || 0)) / delegationPossible) * 100;
+  const selfDelegated = ((node?.ownDelegation || 0) / delegationPossible) * 100;
+  const availableDelegation = ((node?.remainingDelegation || 0) / delegationPossible) * 100;
+  const delegationTotalPages = Math.ceil(delegationCount / 20);
+  const offsetIndex = delegationPage === 1 ? 0 : 20 * (delegationPage - 1);
+
+  if (!node) {
+    return null;
+  }
+  return (
+    <div className="runnode">
+      {delegateToNode && (
+        <Modal
```

Comment:
I think we should create small components for these modals so we don't copy them everywhere we need delegation and undelegation.